File: javascript/introduction/swap.js
//take input from the users
let a = prompt('Enter the first variable: ');
let b = prompt('Enter the second variable: ');

// 1. Using temporary variable
//create a temporary variable
let temp;

//swap variables
temp = a;
a = b;
b = temp;

// 2. Using destructuring assignment
[a, b] = [b, a];

// 3. Using arithmetic operators
// addition and subtraction operator
a = a + b;
b = a - b;
a = a - b;

// 4. Using XOR
// XOR operator
a = a ^ b
b = a ^ b
a = a ^ b

console.log(`The value of a after swapping: ${a}`);
console.log(`The value of b after swapping: ${b}`);

File: cpp/introduction/swap.cpp
#include <iostream>
using namespace std;

int main()
{
    int a = 5, b = 10, temp;

    cout << "Before swapping." << endl;
    cout << "a = " << a << ", b = " << b << endl;

    // 1. Using temporary variable
    temp = a;
    a = b;
    b = temp;

    // 2. Without temporary
    // a = a + b;
    // b = a - b;
    // a = a - b;

    cout << "\nAfter swapping." << endl;
    cout << "a = " << a << ", b = " << b << endl;

    return 0;
}